wasm/main.js

```diff
@@ -87,6 +91,45 @@
     document.addEventListener('keyup', forwardKeyEvent);
   }
 
+  function initializeTouchEventListners() {
+    const keyMappings = [
+      { id: 'key-up', key: 'ArrowUp', keyCode: 38 },
+      { id: 'key-left', key: 'ArrowLeft', keyCode: 37 },
+      { id: 'key-down', key: 'ArrowDown', keyCode: 40 },
+      { id: 'key-right', key: 'ArrowRight', keyCode: 39 },
+      { id: 'key-sp', key: ' ', keyCode: 32 },
+      { id: 'key-esc', key: 'Escape', keyCode: 27 },
+    ]
+    keyMappings.forEach(({id, key, keyCode}) => {
+      const keyButton = document.getElementById(id);
+      keyButton.addEventListener('mousedown', (e) => {
+        handleKeyEvent(key, keyCode, 'keydown');
+      });
+      // Note we have to handle both up and leave to release button
+      keyButton.addEventListener('mouseup', (e) => {
+        handleKeyEvent(key, keyCode, 'keyup');
+      });
+      keyButton.addEventListener('mouseleave', (e) => {
+        handleKeyEvent(key, keyCode, 'keyup');
+      });
+    });
+
+    const toggleButton = document.getElementById('toggle-keyboard');
+    toggleButton.addEventListener('click', (e) => {
+      toggleKeyboard(toggleButton);
+    });
+  }
+
+  function toggleKeyboard(button) {
+    showKeyboard = !showKeyboard;
+    const changeToLabel = showKeyboard ? '⌨︎ ON' : '⌨︎ OFF';
```

Comment:
can we show just the keyboard?

Reply:
We could might be hard to understand. Maybe there's an eye open/closed glyph i can use next to it?